### PR TITLE
EVG-5322 Terminate intent hosts earlier in host termination job

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: units/host_termination.go [hunk 1]
```diff
@@ -199,6 +199,23 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
 		}))
 
 		if !util.StringSliceContains(evergreen.UphostStatus, j.host.Status) {
+			if err := j.host.Terminate(evergreen.User); err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
```
Comment (john-m-liu) [inline units/host_termination.go]: j.AddError instead?

File: units/host_termination.go [hunk 2]
```diff
@@ -199,6 +199,23 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
...
+					"host":     j.host.Id,
+					"provider": j.host.Distro.Provider,
+					"job_type": j.Type().Name,
+					"job":      j.ID(),
+					"message":  "problem terminating instance",
+				}))
+				return
+			}
+			grip.Info(message.Fields{
```
Comment (john-m-liu) [inline units/host_termination.go]: It doesn't seem necessary to log here

File: units/host_termination.go (hunk 1 above)
Comment (bsamek) [reply units/host_termination.go]: done

File: units/host_termination.go (hunk 2 above)
Comment (bsamek) [reply units/host_termination.go]: done

**Pull request merged**